lib/angstrom.ml

```diff
@@ -277,19 +277,19 @@ let rec peek_char_fail =
         peek_char_fail.run input' pos' more' fail succ in
       ensure_suspended 1 input pos more fail succ' }
 
-let satisfy f =
+let satisfy ?(failure_msg="satisfy") f =
```

Comment:
This is going to allocate on every call, which is problematic if you use it below a bind.

You could accomplish the same thing by just doing `satisfy pred <?> "my-pred"` which would generate an error like `my-pred: satisfy <char>`, rather than what you're proposing which is`my-pred: <char>`.

So performance aside it doesn't seem like the optional argument is necessary here.

Reply:
Ok, corrected.